lib/linter.js

```diff
@@ -131,13 +131,14 @@ class Linter {
     let failures = [];
     let rules = [];
 
-    let configuredRules = this.config.rules;
+    let fileConfig = getConfigForFile(this.config, options.filePath);
+    let { loadedRules, rules: configuredRules } = fileConfig;
```

Comment:
I'd rather avoid this destructuring (yeah, I know, but I can't reason about destructuring with aliases 🤕), would you mind updating the references below to use `fileConfig.rules` or `fileConfig.loadedRules`?